src/planner/expand_hypertable.c

```diff
@@ -1060,7 +1060,10 @@ ts_plan_expand_hypertable_chunks(Hypertable *ht, PlannerInfo *root, RelOptInfo *
 	for (unsigned int i = 0; i < num_chunks; i++)
 	{
 		if (!chunks[i]->fd.osm_chunk || include_osm)
-			inh_oids = lappend_oid(inh_oids, chunks[i]->table_id);
+		{
+			inh_chunks = lappend(inh_chunks, chunks[i]);
+		}
+
```

Comment:
Let's move this logic into `get_chunks`? It has a special code path for OSM chunk already anyway. Not sure why we even had the `inh_oids` array before your original change, looks like it was completely redundant with `chunk`, maybe I forgot to remove it during refactoring.

Reply:
done